mlflow/store/model_registry/sqlalchemy_store.py

```diff
@@ -205,6 +206,51 @@ def list_registered_models(self):
             return [sql_registered_model.to_mlflow_entity()
                     for sql_registered_model in session.query(SqlRegisteredModel).all()]
 
+    def search_registered_models(self, filter_string):
+        """
+        Search for registered models in backend that satisfy the filter criteria.
+
+        :param filter_string: A filter string expression. Currently supports a single filter
+                              condition either name of model like ``name = 'model_name'``
+
+        :return: List of :py:class:`mlflow.entities.model_registry.RegisteredModel` objects.
+        """
+        parsed_filter = SearchUtils.parse_filter_for_model_registry(filter_string)
+        if len(parsed_filter) == 0:
+            conditions = []
+        elif len(parsed_filter) == 1:
+            filter_dict = parsed_filter[0]
+            if filter_dict["comparator"] not in \
+                    SearchUtils.VALID_REGISTERED_MODEL_SEARCH_COMPARATORS:
+                raise MlflowException('Search registered models filter expression only '
+                                      'supports the equality(=) comparator, case-sensitive'
+                                      'partial match (LIKE), and case-insensitive partial '
+                                      'match (ILIKE). Input filter string: %s' % filter_string,
+                                      error_code=INVALID_PARAMETER_VALUE)
+            if filter_dict["key"] == "name":
+                if filter_dict["comparator"] == "LIKE":
+                    conditions = [SqlRegisteredModel.name.like(filter_dict["value"])]
+                elif filter_dict["comparator"] == "ILIKE":
+                    conditions = [SqlRegisteredModel.name.ilike(filter_dict["value"])]
+                else:
+                    conditions = [SqlRegisteredModel.name == filter_dict["value"]]
+            else:
+                raise MlflowException('Invalid filter string: %s' % filter_string,
+                                      error_code=INVALID_PARAMETER_VALUE)
+        else:
+            supported_ops = f'"name ' \
+                f'{"/".join(SearchUtils.VALID_REGISTERED_MODEL_SEARCH_COMPARATORS)} ' \
+                f'\'<model_name>\''
```

Comment:
Acceptable filter message will look something like this -- 

`"name" =/LIKE/ILIKE 'model_name'"`

Super confusing. The way you had it before was simpler to understand.

Reply:
The way I had it before was something like
```
"name" (=)(LIKE)(ILIKE) 'model_name'"
```
Do you think that looks better?

Reply:
I think this is a good, clear error message.

Reply:
Thanks for your feedback @andreakress!